Fails to build on Freebsd

When trying to update the rust coreutils, it fails with:

```

      Checking fundu v0.5.1
  error[E0609]: no field `domainname` on type `libc::utsname`
     --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/platform-info-2.0.0/src/platform/unix.rs:121:84
      |
  121 |             debug_struct = debug_struct.field("domainname", &oss_from_cstr(&self.0.domainname));
      |                                                                                    ^^^^^^^^^^ unknown field
      |
      = note: available fields are: `sysname`, `nodename`, `release`, `version`, `machine`
  
  error[E0609]: no field `domainname` on type `libc::utsname`
     --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/platform-info-2.0.0/src/platform/unix.rs:146:38
      |
  146 |             equal = equal && (self.0.domainname == other.0.domainname)
      |                                      ^^^^^^^^^^ unknown field
      |
      = note: available fields are: `sysname`, `nodename`, `release`, `version`, `machine`
  
  error[E0609]: no field `domainname` on type `libc::utsname`
     --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/platform-info-2.0.0/src/platform/unix.rs:146:60
      |
  146 |             equal = equal && (self.0.domainname == other.0.domainname)
      |                                                            ^^^^^^^^^^ unknown field
      |

```

https://github.com/uutils/coreutils/actions/runs/5073973837/jobs/9113647713?pr=4897

cc @rivy

--- a/src/platform/unix.rs
+++ b/src/platform/unix.rs
@@ -116,7 +116,14 @@ impl Debug for UTSName {
             .field("release", &oss_from_cstr(&self.0.release))
             .field("version", &oss_from_cstr(&self.0.version))
             .field("machine", &oss_from_cstr(&self.0.machine));
-        #[cfg(not(target_vendor = "apple" /* or `target_os = "macos"` */))]
+        // BSD-like platforms are missing the domainname field
+        #[cfg(not(any(
+            target_os = "macos",
+            target_os = "dragonfly",
+            target_os = "freebsd",
+            target_os = "openbsd",
+            target_os = "netbsd"
+        )))]
         {
             debug_struct = debug_struct.field("domainname", &oss_from_cstr(&self.0.domainname));
         }
@@ -141,7 +148,14 @@ impl PartialEq for UTSName {
                 other.0.version,
                 other.0.machine,
             );
-        #[cfg(not(target_vendor = "apple" /* or `target_os = "macos"` */))]
+        // BSD-like platforms are missing the domainname field
+        #[cfg(not(any(
+            target_os = "macos",
+            target_os = "dragonfly",
+            target_os = "freebsd",
+            target_os = "openbsd",
+            target_os = "netbsd"
+        )))]
         {
             equal = equal && (self.0.domainname == other.0.domainname)
         }

